### ISSUE TITLE
Multiple "no overloaded function" errors when trying to compile

### ISSUE THREAD

**Issue by uriba107** [EvenAR/node-simconnect#6]
I'm trying to compile for 64bit with p3d v4.5 SDK. I get a lot of "not overloaded function" erros.

As I'm fairly new to Node, and practically clueless with this compilation process on windows, I was hoping you might have some insight.

I'm using master branch code and It have taken me a few hours just to get rid of all the windows compiler errors.

here are just a few examples.
addon.cc(328,50): error C2661:  'v8::Value::ToString': no overloaded function takes 0 arguments
addon.cc(437,58): error C2660:  'v8::Value::Int32Value': function does not take 0 arguments
node-simconnect\src\addon.cc(511,29): error C2512:  'v8::String::Utf8Value::Utf8Value': no appropriate default constructor available

Thanks,
Uri

**Comment by EvenAR** [comment]
Hi, @uriba107. Which version of node.js are you using? These errors are due to some breaking changes in later versions of the v8/Node API. Specifying an older target version should fix this. For example `node-gyp rebuild --target=10.16.3`. Alternatively you can checkout the `upgrades`-branch where I have updated the code to work with node.js 12.x.

**Comment by uriba107** [comment]
Thanks for the response, it's 12.x.

I'll have a look at the branch as suggested. Thanks!

**Comment by BradleyDeLar** [comment]
I'm seeing a number of issues also, even when trying the upgrades branch.

**Comment by BradleyDeLar** [comment]
Amongst numerous deprecated warnings I've narrowed the errors down to the following:
```
SimConnect.lib(SimConnectClient.obj) : error LNK2038: mismatch detected for '_MSC_VER': value '1800' doesn't match va lue '1900' in addon.obj [C:\Users\bradp\Desktop\node-simconnect\node_modules\node-simconnect\build\node-simconnect.vc xproj]
SimConnect.lib(commClient.obj) : error LNK2038: mismatch detected for '_MSC_VER': value '1800' doesn't match value '1 900' in addon.obj [C:\Users\bradp\Desktop\node-simconnect\node_modules\node-simconnect\build\node-simconnect.vcxproj]
SimConnect.lib(SimConnectClient.obj) : error LNK2019: unresolved external symbol "__declspec(dllimport) char const *
__cdecl std::_Winerror_map(int)" (__imp_?_Winerror_map@std@@YAPBDH@Z) referenced in function "public: virtual class s td::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > __thiscall std::_System_error_catego ry::message(int)const " (?message@_System_error_category@std@@UBE?AV?$basic_string@DU?$char_traits@D@std@@V?$allocato r@D@2@@2@H@Z) [C:\Users\bradp\Desktop\node-simconnect\node_modules\node-simconnect\build\node-simconnect.vcxproj]
SimConnect.lib(commClient.obj) : error LNK2019: unresolved external symbol __imp___vsnprintf referenced in function " long __cdecl StringCbPrintfA(char *,unsigned int,char const *,...)" (?StringCbPrintfA@@YAJPADIPBDZZ) [C:\Users\bradp\ Desktop\node-simconnect\node_modules\node-simconnect\build\node-simconnect.vcxproj]
C:\Users\bradp\Desktop\node-simconnect\node_modules\node-simconnect\build\Release\node-simconnect.node : fatal error
LNK1120: 2 unresolved externals [C:\Users\bradp\Desktop\node-simconnect\node_modules\node-simconnect\build\node-simco nnect.vcxproj]
```

**Comment by uriba107** [comment]
@EvenAR 
Followed you instructions, compiled with 12.13.0 no problem 
However I'm getting 
`node-simconnect: Node.js 64 bit is not supported. Please install Node.js 32 bit.`

even though I took the files from p3d sdk and compiled with x64 node.js, I've went as far as trying
`node-gyp rebuild --target=12.13.0 --arch=x64`

any tips for success?

thank you very much for all the hard work.

**Comment by uriba107** [comment]
@EvenAR ,
can this issue be closed now?

**Comment by EvenAR** [comment]
Closing this as the original problem is solved in #7.
